### ISSUE TITLE
Exception:  Key has expired for R Tools on Ubuntu 19.04

### ISSUE THREAD

**Issue by kfish610** [MikhailArkhipov/vscode-r#30]
```
[Info  - 2:02:45 PM] Available R interpreters:
[Info  - 2:02:45 PM] 	[0] CRAN R '3.6.0-2disco'
[Info  - 2:02:45 PM] You can specify the desired interpreter index in the R settings
[Info  - 2:02:45 PM] Selected interpreter: [0] CRAN R '3.6.0-2disco'.

[Info  - 2:02:45 PM] Starting R Process with /usr/lib/R...
[Error - 2:02:47 PM] Connecting to R Workspace failed.
Reason: R session process did not start on the machine 'VSCR'. Exception:  Key has expired
```

**Comment by Hong-Revo** [comment]
Yep, I'm getting this problem too, on trying to run RTools with the nifty new VSCode remote feature. The remote is an Ubuntu 16.04 DSVM, with MRO 3.4.3 installed.

**Comment by MikhailArkhipov** [comment]
What happens if you try running `Microsoft.R.Host` manually? It should be in the `Host` folder.

**Comment by MikhailArkhipov** [comment]
Might be that X509 certificate that SSL key broker process is looking up in the local store has expired.

**Comment by Hong-Revo** [comment]
This is what I got: 

```
hong@hongsdev:~/.vscode-server/extensions/mikhail-arkhipov.r-0.0.6/ls/Host/Linux$ ./Microsoft.R.Host
./Microsoft.R.Host: error while loading shared libraries: libzip.so.4: cannot open shared object file: No such file or directory
```

🤔

**Comment by Hong-Revo** [comment]
Trying to install libzip.so, I ran

```
sudo apt-get update
sudo apt-get upgrade  # possibly a mistake
sudo apt-get install libzip-dev
```

After that, running Microsoft.R.Host caused a segfault.

**Comment by kfish610** [comment]
@Hong-Revo Try using the libzip4 package

**Comment by Hong-Revo** [comment]
libzip4 came along with libzip-dev. I uninstalled libzip-dev and tried again; still segfaulted.

**Comment by Hong-Revo** [comment]
Tried again with a brand-new DSVM. I installed .Net Core, then tried starting R Tools, which produced the "key expired" error. Then I installed libzip4, tried starting R Tools again, and this time it produced an "unknown error". Running Microsoft.R.Host from the command line segfaulted, like before.

**Comment by MikhailArkhipov** [comment]
OK, I tried Docker container created with RTVS on 2019 and it appears to work. DSVM - not completely sure who owns the image, they may need to upgrade the TSL/SSL key there.
@kfish610 - I assume this is local connect.

**Comment by MikhailArkhipov** [comment]
Default key lifetime is 1 year. You may want to look at https://docs.microsoft.com/en-us/visualstudio/rtvs/setting-up-remote-r-workspaces?view=vs-2017 on how to generate new cert for the SSL

**Comment by Hong-Revo** [comment]
Hmm, is it possible to remove the signing requirement entirely? What security risk is it supposed to be mitigating anyway?

**Comment by MikhailArkhipov** [comment]
It is kind of deep embedded into the remoting protocol since it is shared for local, corporate and internet/cloud cases. Remote broker does not have plain http at all, it is configured for HTTPS exclusively.

**Comment by Hong-Revo** [comment]
This may not be architecturally possible, but maybe the VSCode package could treat all connections as local? The remoting feature seems to handle that scenario pretty well -- as far as I can tell, my other extensions work pretty seamlessly.

**Comment by danhalligan** [comment]
I believe I'm seeing a very similar error (`~/.vscode-server/extensions/mikhail-arkhipov.r-0.0.6/ls/Host/Linux/Microsoft.R.Host` segfaults on the remote machine). I'd be interested to know if updating the certificate helped, and if you have any more details on how to do this.

In my case I'm running ubuntu 18.04 as both the local and remote machines.

**Comment by hmassalha** [comment]
same problem 
do you think contacting the IT unit will solve this issue? is that something related to opening/allowing graphical interface retrieval?

**Comment by DanielReid** [comment]
> ```
> [Info  - 2:02:45 PM] Available R interpreters:
> [Info  - 2:02:45 PM] 	[0] CRAN R '3.6.0-2disco'
> [Info  - 2:02:45 PM] You can specify the desired interpreter index in the R settings
> [Info  - 2:02:45 PM] Selected interpreter: [0] CRAN R '3.6.0-2disco'.
> 
> [Info  - 2:02:45 PM] Starting R Process with /usr/lib/R...
> [Error - 2:02:47 PM] Connecting to R Workspace failed.
> Reason: R session process did not start on the machine 'VSCR'. Exception:  Key has expired
> ```

I have exactly this issue, using the extension *locally* on Ubuntu 19.10.

What is the 'VSCR' machine? Should I be updating a key somewhere manually? I'm not experienced with this ecosystem, apologies.

**Comment by MikhailArkhipov** [comment]
You can try and have a look at https://docs.microsoft.com/en-us/visualstudio/rtvs/setting-up-remote-r-workspaces?view=vs-2017 which explains how to set up TSL/SSL key. My guess is that keys provided when RTVS was released have expired. The connection between the editor/tools and process that loads actual R interpreter is over TSL even locally.

**Comment by MikhailArkhipov** [comment]
I can drop `https` to `http` via settings though. It may not be a good idea long term for remote sessions though.

**Comment by MikhailArkhipov** [comment]
Fixed in 0.0.10
